extensions-contrib/rocketmq-indexing-service/src/main/java/org/apache/druid/indexing/rocketmq/RocketMQRecordSupplier.java

```diff
@@ -0,0 +1,243 @@
+      messageQueues.add(mq);
+    }
+
+    if (!messageQueues.isEmpty()) {
+      consumer.assign(messageQueues);
+    }
+
+    try {
+      if (!consumer.isRunning()) {
+        consumer.start();
+      }
+    }
+    catch (MQClientException e) {
+      log.error(e.getErrorMessage());
+    }
+    clientInstance = MQClientManager.getInstance().getOrCreateMQClientInstance(consumer);
+  }
+
+  @Override
+  public void seek(StreamPartition<String> partition, Long sequenceNumber)
+  {
+    MessageQueue mq = PartitionUtil.genMessageQueue(partition.getStream(), partition.getPartitionId());
+    try {
+      consumer.seek(mq, sequenceNumber);
+    }
+    catch (MQClientException e) {
+      log.error(e.getErrorMessage());
+      consumer.getOffsetStore().updateOffset(mq, sequenceNumber, false);
+    }
+  }
+
+  @Override
+  public void seekToEarliest(Set<StreamPartition<String>> partitions)
+  {
+    for (StreamPartition<String> partition : partitions) {
+      MessageQueue mq = PartitionUtil.genMessageQueue(partition.getStream(), partition.getPartitionId());
+      try {
+        consumer.seekToBegin(mq);
+      }
+      catch (MQClientException e) {
+        log.error(e.getErrorMessage());
+      }
+    }
+  }
+
+  @Override
+  public void seekToLatest(Set<StreamPartition<String>> partitions)
+  {
+    for (StreamPartition<String> partition : partitions) {
+      MessageQueue mq = PartitionUtil.genMessageQueue(partition.getStream(), partition.getPartitionId());
+      try {
+        consumer.seekToEnd(mq);
+      }
+      catch (MQClientException e) {
+        log.error(e.getErrorMessage());
+      }
+    }
+  }
+
+  @Override
+  public Set<StreamPartition<String>> getAssignment()
+  {
+    if (this.streamPartitions == null) {
+      return new HashSet<StreamPartition<String>>();
+    }
+    return this.streamPartitions;
+  }
+
+  @Nonnull
+  @Override
+  public List<OrderedPartitionableRecord<String, Long, RocketMQRecordEntity>> poll(long timeout)
+  {
+    List<OrderedPartitionableRecord<String, Long, RocketMQRecordEntity>> polledRecords = new ArrayList<>();
+    for (MessageExt messageExt : consumer.poll(timeout)) {
+
+      polledRecords.add(new OrderedPartitionableRecord<>(
+          messageExt.getTopic(),
+          PartitionUtil.genPartition(messageExt.getBrokerName(), messageExt.getQueueId()),
+          messageExt.getQueueOffset(),
+          messageExt.getBody() == null ? null : ImmutableList.of(new RocketMQRecordEntity(messageExt.getBody()))
+      ));
+    }
+    return polledRecords;
+  }
+
+  @Override
+  public Long getLatestSequenceNumber(StreamPartition<String> partition)
+  {
+    long offset = Long.MAX_VALUE;
+    try {
+      offset = clientInstance.getMQAdminImpl().maxOffset(PartitionUtil.genMessageQueue(partition.getStream(), partition.getPartitionId()));
+    }
+    catch (MQClientException e) {
+      log.error(e.getErrorMessage());
```

Comment:
if there's exception, returning would be Long.MAX_VALUE, is that expected ?

Reply:
I check code and find this method is mainly used to get offset when use "consume from latest". So if can not connect to  broker, this task will be interrupted. If catch MQClientException, it means this message queue not exists or other trouble,I think reset to Long.MAX_VALUE will not success in this condition. I think is safe to return it. Can I do something if catch exception like interrupt this task? I do not know how to throw an exception in the method.